HoloJS/HoloJsHost/RenderingContext2D.cpp

```diff
@@ -50,62 +54,283 @@ void RenderingContext2D::drawImage(HologramJS::API::ImageElement* imageElement,
 }
 
 void RenderingContext2D::clearRect(Rect& rect)
 {
-    CanvasDrawingSession ^ session = m_canvasRenderTarget->CreateDrawingSession();
-    session->Blend = CanvasBlend::Copy;  // Overwrite everything!
+    // Save the previous blend value
+    CanvasBlend blend = m_session->Blend;
+
+    m_session->Blend = CanvasBlend::Copy;
 
     Color color;
     color.A = 0;
     color.R = 0;
     color.G = 0;
     color.B = 0;
 
-    session->FillRectangle(rect, color);
-    session = nullptr;
+    m_session->FillRectangle(rect, color);
+
+    m_session->Blend = blend;
 }
 
 void RenderingContext2D::fillRect(Rect& rect, Color& color)
 {
-    CanvasDrawingSession ^ session = m_canvasRenderTarget->CreateDrawingSession();
-    session->FillRectangle(rect, color);
-    session = nullptr;
+    Color cm = color;
+    cm.A *= m_globalOpacity;
+    m_session->FillRectangle(rect, cm);
 }
 
 void RenderingContext2D::fillRectGradient(Rect& rect,
                                           float2& start,
                                           float2& end,
                                           Platform::Array<CanvasGradientStop> ^ stops)
 {
-    CanvasDrawingSession ^ session = m_canvasRenderTarget->CreateDrawingSession();
     auto brush = ref new CanvasLinearGradientBrush(m_canvasRenderTarget, stops);
     brush->StartPoint = start;
     brush->EndPoint = end;
-    session->FillRectangle(rect, brush);
-    session = nullptr;
+    m_session->FillRectangle(rect, brush);
 }
 
-void RenderingContext2D::fillText(
-    std::wstring& text, float2& point, Color& color, int fontSize, std::wstring& fontFamily)
+void RenderingContext2D::fillText(std::wstring& text,
+                                  float2& point,
+                                  Color& color,
+                                  int fontSize,
+                                  std::wstring& fontFamily,
+                                  std::wstring& fontWeight,
+                                  std::wstring& fontStyle,
+                                  int alignment)
 {
-    CanvasDrawingSession ^ session = m_canvasRenderTarget->CreateDrawingSession();
+    CanvasTextFormat ^ format = ref new CanvasTextFormat();
+    format->FontFamily = ref new Platform::String(fontFamily.c_str());
+    if (fontWeight == L"bold") {
+        format->FontWeight = Windows::UI::Text::FontWeights::Bold;
+    }
+    if (fontStyle == L"italic") {
+        format->FontStyle = Windows::UI::Text::FontStyle::Italic;
+    }
+    format->FontSize = fontSize;
+    if (alignment == 0) {
+        format->HorizontalAlignment = CanvasHorizontalAlignment::Left;
+    }
+    if (alignment == 1) {
+        format->HorizontalAlignment = CanvasHorizontalAlignment::Center;
+    }
+    if (alignment == 2) {
+        format->HorizontalAlignment = CanvasHorizontalAlignment::Right;
+    }
+    format->VerticalAlignment = CanvasVerticalAlignment::Bottom;
+    format->LineSpacing = 1;
```

Comment:
It's always going to be single line - HTML5 canvas doesn't support line breaks (and I believe we need to strip those to be compliant with the standard). I set this because I tried to produce the correct vertical alignment, in which the "y" should specify the baseline y position of the text.